Inferences/InterpretedEvaluation.cpp

```diff
@@ -112,25 +110,17 @@ Clause* InterpretedEvaluation::simplify(Clause* cl)
         }
       }
       
-      newLits[next++]=res;
+      resLits->push(res);
     }
     if(!modified) {
       return cl;
     }
 
-    ASS(sideConditions.isEmpty())
-    Stack<Literal*>::Iterator side(sideConditions);
-    newLits.expand(clen+sideConditions.length());
-    while(side.hasNext()){ newLits[next++]=side.next();}
-    int newLength = next;
-    Clause* res = new(newLength) Clause(newLength,SimplifyingInference1(InferenceRule::EVALUATION, cl));
-
-    for(int i=0;i<newLength;i++) {
-      (*res)[i] = newLits[i];
-    }
+    ASS(sideConditions->isEmpty())
+    resLits->loadFromIterator(sideConditions->iterFifo());
```

Comment:
I know it looks like it was there already in master, but does it make sense to follow:
```
ASS(sideConditions->isEmpty())
```
by loading the (empty) side conditions to `resLits`?

Reply:
haha no this actually doesn't make sense. I think i remember what was going on here. There used to be the possibility of introducing sed conditions in interpreted evaluation (before I even start to work on theory reasoning code). And for some reason I thought that side condition introduction was dead code, so I added the assertion. Turns out it is indeed, if we never triggered the assertion yet. 

Let me quickly get rid of the dead code.

Reply:
done

Reply:
Nice, thanks!